### ISSUE TITLE
Port twisted.internet.interfaces to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5942]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5942|
|Type|enhancement|
|Created|2012-09-05 15:22:57Z|
|Branch|https://github.com/twisted/twisted/tree/interfaces-python3-5942|

Most stuff in Twisted depends on this module.  And conveniently, it has no behavior and no unit tests, so it just needs to be syntactically valid.

<details><summary>Searchable metadata</summary>

```
trac-id__5942 5942
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_interfaces_python3_5942 branches/interfaces-python3-5942
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1346858577000000 1346858577000000
changetime__1346954164000000 1346954164000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [35508]) Branching to 'interfaces-python3-5942'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [35515]) Change to Python 3 compatible syntax and get rid of the two deprecated interfaces (to avoid needing to think about what to do about deprecatedModuleAttribute)

refs [#5942](https://github.com/twisted/twisted/issues/5942)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [35523]) Okay, so there were tests, but there are not anymore.

refs [#5942](https://github.com/twisted/twisted/issues/5942)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @itamarst|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/interfaces-python3-5942)

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_5"></a>|@itamarst set owner to @exarkun|
|-|-|

  1. There were some epoll failures... I think those are unrelated, though, I can't see how they'd happen because of this change, nor how they'd happen only on that build slave. You should check too just in case.
  2. Add `twisted.internet` and `twisted.internet.interfaces` to `_twistedpython3.py`.

Then merge. Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [35539]) Merge interfaces-python3-5942

Author: exarkun
Reviewer: itamar
Fixes: [#5942](https://github.com/twisted/twisted/issues/5942)
Fixes: [#5943](https://github.com/twisted/twisted/issues/5943)
Fixes: [#5944](https://github.com/twisted/twisted/issues/5944)

Port `twisted.internet.interfaces` to Python 3.  To simplify this, drop the two deprecated
interfaces it previously offered, `IReactorArbitrary` and `IFinishableConsumer`.  Also
delete the deprecated services in `twisted.application.internet` which were based on
`IReactorArbitrary`.
